lib/pure/typetraits.nim

```diff
@@ -167,6 +167,16 @@ since (1, 3, 5):
 
     typeof(block: (for ai in a: ai))
 
+since (1, 5, 1):
+  template typeOrVoid*[T](a: T): type =
+    ## Returns the type of `a` or `void` if it's a statement, working around
+    ## a bug in the compiler which doesn't support `stmt is void`.
```

Comment:
Can't we fix the compiler bug instead?

Reply:
=> https://github.com/nim-lang/Nim/pull/17807